libp2p/pubsub/pubsub_notifee.py

```diff
@@ -29,7 +30,9 @@ def __init__(
         can process dead peers after we disconnect from each other
         """
         self.initiator_peers_queue = initiator_peers_queue
+        self.initiator_peers_queue_lock = trio.Lock()
         self.dead_peers_queue = dead_peers_queue
+        self.dead_peers_queue_lock = trio.Lock()
```

Comment:
I'm confused why these locks are needed...  don't the channels handle this for us?  Maybe deserves a code comment explaining why the lock is necessary.

Reply:
Please correct me if I'm wrong: my thought is that `PubsubNotifee.connected` is possibly called multiple times by `Swarm` whenever there are new connections and it's possible for `self.initiator_peers_queue.send` being called concurrently and `trio.BusyResourceError` is raised.

Reply:
Update: I removed the lock and changed the channel to be of size `math.inf`, to avoid `channel.send` from being blocked and prevent `BusyResourceError` from happening.

Reply:
I don't think using `math.inf` for channel size is a good idea.  Channels are sized to 1) ensure that they cannot grow to some indefinite size and 2) to provide [backpressure](https://www.techopedia.com/definition/24131/backpressure) when the send side of the channel is operating faster than the receive side can consume.

Setting to `math.inf` effectively disables both of these mechanisms which doesn't seem like a good trade-off.

Reply:
Alright, I agree with your point that`math.inf` is not suitable for `initiator_peers_queue` and `dead_peers_queue` here since the handle loop should be capable of catching up with the speed of peer churns. It seems to me that to make both sides(send and receive) synchronous I will need to add the locks back to prevent `BusyResourceError` from happening. Do you think it makes sense to add the locks back?

Reply:
According to the docs: 

https://trio.readthedocs.io/en/latest/reference-core.html#trio.MemorySendChannel

`MemorySendChannel` doesn't need a lock and allows multiple tasks to call `send` at the same time so I think you're safe without the lock.

Reply:
My fault. I'm changing it back. Thank you for the reference.